addons/web/static/src/webclient/clickbot/clickbot.js

```diff
@@ -385,19 +386,28 @@ async function testViews() {
  *  @returns {Promise}
  */
 async function testMenuItem(element) {
-    const menuDescription = element.innerText.trim() + " " + element.dataset.menuXmlid;
+    const menu = element.dataset.menuXmlid;
+    const menuDescription = element.innerText.trim() + " " + menu;
     browser.console.log(`Testing menu ${menuDescription}`);
-    testedMenus.push(element.dataset.menuXmlid);
-    if (BLACKLISTED_MENUS.includes(element.dataset.menuXmlid)) {
+    testedMenus.push(menu);
+    if (BLACKLISTED_MENUS.includes(menu)) {
         return Promise.resolve(); // Skip black listed menus
     }
     const startActionCount = actionCount;
     await triggerClick(element, `menu item "${element.innerText.trim()}"`);
     try {
         await waitForCondition(() => startActionCount !== actionCount);
-        await testStudio();
-        await testFilters();
-        await testViews();
+        if (MODAL_MENUS.includes(menu)) {
```

Comment:
I don't understand how this can work, because we don't go through updateUI in the modal case, so you should get stuck in the waitForCondition